cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseTable.java

```diff
@@ -243,6 +276,19 @@ private Put getIncrementalPut(Put existing, byte[] row) {
       .build();
   }
 
+  private IncrementBuilder getIncrement(IncrementBuilder existing, byte[] row, boolean transactional)
+    throws IOException {
+    if (existing != null) {
```

Comment:
wow. That means we have had a bug in the existing code... so this needs to be cleared at the start of a transaction.